war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -228,50 +228,73 @@ var FormChecker = {
    * @param url
    *      Remote doXYZ URL that performs the check. Query string should include the field value.
    * @param method
-   *      HTTP method. GET or POST. I haven't confirmed specifics, but some browsers seem to cache GET requests.
+   *     Unused, kept to maintain compatibility with the old signature.
```

Comment:
`matrix-auth` and `azure-ad` call this with "GET". This changes the existing behavior. Not sure there is a valid justification for that.

If there is a valid justification for changing the existing behavior, then this calling convention is pretty ugly. Better to define a new two-argument method and deprecate the three-argument one, logging a warning if anyone is still calling the three-argument version and filing PRs to update consumers.

Reply:
JavaScript doesn't have function overloading.
https://stackoverflow.com/questions/456177/function-overloading-in-javascript-best-practices

We're stuck with that function with that name.
We could create a new function with a new name and use an object for parameters to make it more flexible in future

Reply:
Still not sure there is a valid justification for changing the existing behavior.

Reply:
core 2.285 changed the default to post, these 2 would have been missed by accident as they will be the only consumers calling it directly.

anyway I've added some compat so they will be able to set GET for now and will file separate PRs to switch them to post if that's okay?